message/infrastructure/io/subscriber.go

```diff
@@ -0,0 +1,215 @@
+package io
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+)
+
+type SubscriberConfig struct {
+	BufferSize       int
```

Comment:
some doc here will be useful

Reply:
done b3ae9f252ccb6e90efcf9f72512765a9df71cff4